### PR TITLE
Fallback to db.connection_string for sql backend url

### PR DESCRIPTION
**Pull request by rish691**

### CODE REVIEW

File: span-normalizer/span-normalizer/src/main/java/org/hypertrace/core/spannormalizer/fieldgenerators/SqlFieldsGenerator.java [hunk 1]
```diff
@@ -71,7 +71,16 @@ protected void maybePopulateSqlUrlForOtelSpan(
       final Map<String, AttributeValue> attributeFieldMap) {
     if (DbSemanticConventionUtils.isSqlTypeBackendForOtelFormat(attributeFieldMap)) {
       Optional<String> sqlUrl = DbSemanticConventionUtils.getBackendURIForOtelFormat(attributeFieldMap);
-      sqlUrl.ifPresent(s -> eventBuilder.getSqlBuilder().setUrl(s));
+      if (sqlUrl.isPresent()) {
```
Comment (kotharironak) [inline span-normalizer/span-normalizer/src/main/java/org/hypertrace/core/spannormalizer/fieldgenerators/SqlFieldsGenerator.java:74]: Can't we use `getSqlURI` form DbSemanticsConvention?

Comment (rish691) [reply span-normalizer/span-normalizer/src/main/java/org/hypertrace/core/spannormalizer/fieldgenerators/SqlFieldsGenerator.java:74]: That accepts Event object, we have an attribute map here. I can create one more method there. Should I?

**Comment by tim-mwangi** [comment]
Some unit tests?

File: span-normalizer/span-normalizer/src/main/java/org/hypertrace/core/spannormalizer/fieldgenerators/SqlFieldsGenerator.java (hunk 1 above)
Comment (kotharironak) [reply span-normalizer/span-normalizer/src/main/java/org/hypertrace/core/spannormalizer/fieldgenerators/SqlFieldsGenerator.java:74]: Yes, it's a common code, and it's difficult to maintain.

**Comment by kotharironak** [comment]
@rish691 snyk scan is failing.

File: semantic-convention-utils/src/main/java/org/hypertrace/semantic/convention/utils/db/DbSemanticConventionUtils.java [hunk 2]
```diff
@@ -163,7 +163,33 @@ public static boolean isSqlTypeBackendForOtelFormat(
...
       return Optional.of(SpanAttributeUtils.getStringAttribute(event, SQL_URL));
     }
-    return getBackendURIForOtelFormat(event);
+    Optional<String> backendUrl = getBackendURIForOtelFormat(event);
+    if (backendUrl.isPresent()) {
+      return backendUrl;
+    }
+    if (SpanAttributeUtils.containsAttributeKey(
+        event, OTelDbSemanticConventions.DB_CONNECTION_STRING.getValue())) {
+      String url = SpanAttributeUtils.getStringAttribute(
```
Comment (kotharironak) [inline semantic-convention-utils/src/main/java/org/hypertrace/semantic/convention/utils/db/DbSemanticConventionUtils.java:172]: Don't we have to convert this `String` to URI object?
```
try {
new URI(URL)
} catch (Mulformed) {
}
```

**Comment by JBAhire** [comment]
> @rish691 snyk scan is failing.

I guess buchi fixed it here: https://github.com/hypertrace/hypertrace-ingester/pull/87#issuecomment-744271612

File: semantic-convention-utils/src/main/java/org/hypertrace/semantic/convention/utils/db/DbSemanticConventionUtils.java (hunk 2 above)
Comment (rish691) [reply semantic-convention-utils/src/main/java/org/hypertrace/semantic/convention/utils/db/DbSemanticConventionUtils.java:174]: done

**Review by kotharironak — APPROVED** [review]

File: semantic-convention-utils/src/main/java/org/hypertrace/semantic/convention/utils/db/DbSemanticConventionUtils.java [hunk 3]
```diff
@@ -163,7 +165,39 @@ public static boolean isSqlTypeBackendForOtelFormat(
...
+          event, OTelDbSemanticConventions.DB_CONNECTION_STRING.getValue());
+      if (!isValidURI(url)) {
```
Comment (jcchavezs) [inline semantic-convention-utils/src/main/java/org/hypertrace/semantic/convention/utils/db/DbSemanticConventionUtils.java:176]: nit, if you do `isValirURI(url) { return Optional.of(url); }` I think you can just keep the last `return Optional.empty()`.

File: semantic-convention-utils/src/test/java/org/hypertrace/semantic/convention/utils/db/DbSemanticConventionUtilsTest.java [hunk 4]
```diff
@@ -193,6 +194,46 @@ public void testGetSqlURI() {
...
+            SemanticConventionTestUtil.buildAttributeValue(OTelDbSemanticConventions.MYSQL_DB_SYSTEM_VALUE.getValue()),
+            OTelDbSemanticConventions.DB_CONNECTION_STRING.getValue(),
+            SemanticConventionTestUtil.buildAttributeValue("jdbc:mysql://mysql:3306/shop")));
+    when(e.getAttributes()).thenReturn(attributes);
+    v = DbSemanticConventionUtils.getSqlURI(e);
+    assertEquals("jdbc:mysql://mysql:3306/shop", v.get());
+  }
+
+  @Test
+  public void testGetSqlUrlForOtelFormat() {
```
Comment (jcchavezs) [inline semantic-convention-utils/src/test/java/org/hypertrace/semantic/convention/utils/db/DbSemanticConventionUtilsTest.java:210]: I wonder if we need some test where the user/password are included in the URL. Not sure tif that is a thing among jdbc.

**Comment by jcchavezs** [comment]
Late to the party @rish691 but left you a couple of comments.

File: semantic-convention-utils/src/main/java/org/hypertrace/semantic/convention/utils/db/DbSemanticConventionUtils.java (hunk 3 above)
Comment (rish691) [reply semantic-convention-utils/src/main/java/org/hypertrace/semantic/convention/utils/db/DbSemanticConventionUtils.java:176]: Sure, I will do it in a subsequent pr.

File: semantic-convention-utils/src/test/java/org/hypertrace/semantic/convention/utils/db/DbSemanticConventionUtilsTest.java (hunk 4 above)
Comment (rish691) [reply semantic-convention-utils/src/test/java/org/hypertrace/semantic/convention/utils/db/DbSemanticConventionUtilsTest.java:210]: You mean if username/pwd is present we should drop the url?

**Pull request merged**